mypyc/test-data/irbuild-str.test

```diff
@@ -293,20 +293,29 @@ L0:
 def f(s: str) -> None:
     s.encode()
     s.encode('utf-8')
+    s.encode('utf-8', 'strict')
+    s.encode('utf-8', errors='strict')
+    s.encode('utf-8', 'backslashreplace')
+    s.encode(encoding='ascii')
     s.encode('ascii', 'backslashreplace')
```

Comment:
Also test cases where the specialization shouldn't be applied. Examples: `s.encode(x)`, `s.encode('a', x)`, `s.encode('utf8', errors=x)` and `s.encode(errors=x)` where `x` is not a literal.

Test cases where we have two keyword args: `s.encode(encoding=..., errors=...)` and `s.encode(errors=..., encoding=...)`.

Reply:
I've updated the logic to work out the args better and added more tests. Please take another look!